src/docker/restorableappengine.cc

```diff
@@ -794,13 +797,66 @@ void RestorableAppEngine::pullImage(const std::string& client, const std::string
              max_parallel_pulls % format % shared_blob_dir.string() % src % dst_dir.string(),
          "failed to pull image", boost::this_process::environment());
   }
+  const auto index_manifest{dst_dir / "index.json"};
+  const auto index_manifest_desc{Utils::parseJSONFile(index_manifest)};
+  HashedDigest manifest_digest{index_manifest_desc["manifests"][0]["digest"].asString()};
+
+  const auto manifest_file{shared_blob_dir / "sha256" / manifest_digest.hash()};
+  if (!boost::filesystem::exists(manifest_file)) {
+    LOG_DEBUG << "missing App image manifest; image: " << src << "; manifest: " << manifest_file;
+    // return false; TODO throw
+  }
+  const auto manifest{Utils::parseJSONFile(shared_blob_dir / "sha256" / manifest_digest.hash())};
```

Comment:
@doanac Basically, we compose the manifest that is received and parsed here https://github.com/foundriesio/meta-lmp/pull/1288/files#diff-3fece4d083364dc504f7349dc19cc85b5a6b8fceba1ce2a8871ccd9f31827383R16. The manifest is stored inside of a TAT file (`load-manifest.tar`) which then is specified in the `docker load -i <>` command.

Reply:
typo: TAT -> TAR?